lib/cmock_plugin_manager.rb

```diff
@@ -14,14 +16,7 @@ def initialize(config, utils)
     plugins_to_load.each do |plugin|
       plugin_name = plugin.to_s
       object_name = "CMockGeneratorPlugin" + camelize(plugin_name)
-      begin
-        unless (Object.const_defined? object_name)
-          require "#{File.expand_path(File.dirname(__FILE__))}/cmock_generator_plugin_#{plugin_name.downcase}.rb"
```

Comment:
We use CMock in a multi-threaded environment to parallelise generation of mock files. We noticed a bug where we would get an ArgumentError on the 'eval' (2 instead of 0 or something along those lines).

This was happening because the 'require' is not atomic. The Object.const_defined? object_name could have been returning true when a 'require' is not yet fully complete.

I added a mutex around it and the problem went away. Open to a better solution however, using a mutex may not be the best solution.